resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -147,12 +147,15 @@ private[spark] class Client(
           throw e
       }
 
+      val sId = s"${Option(conf.namespace).getOrElse("default")}:" +
```

Comment:
Is "default" always the default, or can it change e.g. depending on the user's `~/.kube/config`?